Server GET

JoelPfammatter98 opened the pull request:
/title/id
 /board/id
 /data

pls Control

JoelPfammatter98 committed:
Server getData

Christian-Seematter committed:
#11 Add package.json

- Small formatting in server.ts

JoelPfammatter98 committed:
neue Ausgaben

Gibt nun genau ein Title mit der mitgegebenen ID zurück
/title/id

Gibt den Titel, anzahlFolgen, rating der mitgegebenen ID zurück
/board/id

JoelPfammatter98 committed:
Merge branch 'server' of https://github.com/bregySilvan/TabSolution into server

Commit:
https://stackoverflow.com/questions/36002493/no-access-control-allow-origin-header-in-angular-2-app

#22 To get the permission to get the data from the server, you need to add a few lines.
Found on https://stackoverflow.com/questions/36002493/no-access-control-allow-origin-header-in-angular-2-app

Commit:
Removed TabSolutionsClient from server branch

JoelPfammatter98 committed:
add ID

JoelPfammatter98 committed:
Merge branch 'server' of https://github.com/bregySilvan/TabSolution into server

JoelPfammatter98 committed:
change /title

Christian-Seematter commented on exemples/server/server.ts:
```diff
@@ -0,0 +1,64 @@
+            }
+        }
+        catch(err) {
+            obj = "ERROR ID ist nicht vorhanden :("
```
Good for now! Later you can return this in a JSON and tell the client which ID(s) failed to fetch data. Imagine when a client makes a lot of requests later on, it might be hard to debug which ones failed, unless the API gives that information.

Christian-Seematter commented on exemples/server/server.ts:
```diff
@@ -0,0 +1,64 @@
+})
+
+
+var server = app.listen(8888, function () {
```
Good for now, this can later be an optional start parameter for the app with a default.

Christian-Seematter commented on exemples/server/server.ts:
```diff
@@ -0,0 +1,64 @@
+
+var server = app.listen(8888, function () {
+    let port = server.address().port
+    console.log("Runnint express at: " + port)
```
Could output the address it's running at too.

Christian-Seematter commented on exemples/server/server.ts:
```diff
@@ -0,0 +1,64 @@
+app.get('/board/:id', function (req, res) {
+    let ID:string = req.params.id;
+    let IDNew = "ID" + ID;
+    fs.readFile( __dirname + "/" + "daten.json", 'utf8', function (err, data) {
```
Good for now! Looking forward to the abstraction of this into a class with methods to read and write the JSON.

Christian-Seematter commented on exemples/server/server.ts:
```diff
@@ -0,0 +1,64 @@
+var express = require('express');
```
Take this out of examples/, this is your project server, not a demo.

Christian-Seematter reviewed (changes requested):
Great work! The only change I require is having this in a proper project location. Should this not be app/server? Coordinate with the team.

JoelPfammatter98 commented:
change Ordner after merge to master

Christian-Seematter commented:
@TheScorpio13 What order are you changing? Is this a task for changes on this PR? Why change it _after_ merging? Then you'll need to create a new PR. You can continue to commit to this branch and the PR gets updated automatically.

bregySilvan commented:
@Christian-Seematter 
I discuessed with scorpio. We merge this PR in and afterwards https://github.com/bregySilvan/TabSolution/pull/18 will get merged in to change folder structure and start commands.

Christian-Seematter commented:
Ok.

@TheScorpio13 Please resolve the conflict. You can do this here on github, by clicking on "resolve conflicts". It looks like server.ts has change since this branch was created, so you need to decide which version should be merged to master now.

bregySilvan committed:
Merge branch 'master' into server

olinator98 commented:
@TheScorpio13 
Please have a look at my latest issue #35 Interface redeclaration. In the branch client is also a version of my server, please have a look at it. The problem which we get with this server is the incompatability with my client. I'm supposed to say that we should use my version of the server, because all the small changes were already done and it is working fine with my version of the client. 
Cherio

bregySilvan committed:
#27 create directory structure

-add location /src/server
-move server to /src/server
-move daten.json to ‘/src/server/var/daten.json’
-add /src/config.ts file
-changed fs to fs-extra module since fs doesn’t exist as standalone
anymore

bregySilvan commented:
@Christian-Seematter 
-Project is moved to right location.
-This is our first server version.
-I tested properly and everything works. 

=> merged
